### PR TITLE
[Merged by Bors] - refactor: golf Normal.of_isSplittingField

### PR DESCRIPTION
**Pull request by alreadydone**
Before: *Construction* only imports *Normal*, which transitively imports *IsSplittingField*
Now: *Normal* imports *Construction*, *Construction* only imports *IsSplittingField*
So no extra transitive import is added to any file other than *Construction* and *Normal*.
As a consequence, `Polynomial.SplittingField.instNormal` is moved from *Construction* to *Normal*.

`adjoin_rootSet_eq_range` is added to *IsSplittingField*.

`splits_of_comp` in *Splits* is extracted from `splits_of_splits` in *IsSplittingField*.

Source of proof: https://math.stackexchange.com/a/2585087/12932

Move `Algebra.adjoin.liftSingleton` from *IsAlgClosed/Basic* to *Adjoin/Field* in order to speed up `lift_of_splits` (renamed to add namespace `Polynomial`).

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/Data/Polynomial/Splits.lean [hunk 1]
```diff
@@ -430,6 +424,16 @@ theorem splits_iff_exists_multiset {f : K[X]} :
     splits_of_exists_multiset i hs⟩
 #align polynomial.splits_iff_exists_multiset Polynomial.splits_iff_exists_multiset
 
+theorem splits_of_comp (j : L →+* F) {f : K[X]} (h : Splits (j.comp i) f)
+    (roots_mem_range : ∀ a ∈ (f.map (j.comp i)).roots, a ∈ j.range) : Splits i f := by
```
Comment (tb65536) [inline Mathlib/Data/Polynomial/Splits.lean:428]: Would it make sense to first prove this theorem without `i` (i.e., set `i = id`) and then deduce this theorem by specializing to `f.map i`?

File: Mathlib/FieldTheory/Normal.lean [hunk 2]
```diff
@@ -150,93 +145,42 @@ theorem AlgEquiv.transfer_normal (f : E ≃ₐ[F] E') : Normal F E ↔ Normal F
...
-      (Algebra.adjoin_res_eq_adjoin_res F E C D hFEp.adjoin_rootSet AdjoinRoot.adjoinRoot_eq_top)
-  rw [Set.image_singleton, RingHom.algebraMap_toAlgebra, AdjoinRoot.lift_root]
+  have hx : IsIntegral F x := isIntegral_of_noetherian (IsNoetherian.iff_fg.2 hFE) x
+  let L := (p * minpoly F x).SplittingField
+  have hL := splits_of_splits_mul' _ ?_ (SplittingField.splits (p * minpoly F x))
+  · let j : E →ₐ[F] L := IsSplittingField.lift E p hL.1
+    refine ⟨hx, splits_of_comp _ (j : E →+* L) ?_ fun a ha ↦ ?_⟩
+    · rw [j.comp_algebraMap]; exact hL.2
+    rw [j.comp_algebraMap, mem_roots ((minpoly.monic hx).map _).ne_zero, IsRoot, eval_map] at ha
+    let FxL : F⟮x⟯ →ₐ[F] L := (adjoin.powerBasis hx).lift a ?_
```
Comment (tb65536) [inline Mathlib/FieldTheory/Normal.lean:165]: Not sure if it's any easier, but another way to construct this map is to use https://leanprover-community.github.io/mathlib4_docs/Mathlib/FieldTheory/Adjoin.html#IntermediateField.algHomAdjoinIntegralEquiv

File: Mathlib/Data/Polynomial/Splits.lean (hunk 1 above)
Comment (alreadydone) [reply Mathlib/Data/Polynomial/Splits.lean:428]: Almost the same proof works for 
```
theorem splits_id_of_splits {f : K[X]} (h : Splits i f)
    (roots_mem_range : ∀ a ∈ (f.map i).roots, a ∈ i.range) : Splits (RingHom.id K) f := sorry
```
but I need an additional `RingHom.comp_id` in the `conv_lhs => rw` block. So I'd rather deduce it from the `comp` version.

File: Mathlib/FieldTheory/Normal.lean (hunk 2 above)
Comment (alreadydone) [reply Mathlib/FieldTheory/Normal.lean]: Yeah it requires adding another lemma [124c3d5](https://github.com/leanprover-community/mathlib4/pull/8004/commits/124c3d54b9e84cb75fddb3a286372e6d19066136). I think the long term solution is to switch to [IsAdjoinRoot](https://leanprover-community.github.io/mathlib4_docs/Mathlib/RingTheory/IsAdjoinRoot.html#IsAdjoinRoot) so as to have a unified lemma [IsAdjoinRoot.lift_root](https://leanprover-community.github.io/mathlib4_docs/Mathlib/RingTheory/IsAdjoinRoot.html#IsAdjoinRoot.lift_root) without duplication.

**Comment by riccardobrasca** [comment]
Nice golf, thanks!

bors merge

**Pull request closed without merging**